cmd/pinshare-tray/tray.go

```diff
@@ -0,0 +1,686 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"pinshare/internal/winservice"
+
+	"github.com/getlantern/systray"
+	"golang.org/x/sys/windows"
+)
+
+const (
+	// healthCheckTimeout is the HTTP timeout for health check requests
+	healthCheckTimeout = 2 * time.Second
+
+	// serviceActionDelay is the delay after starting/stopping service before checking status
+	serviceActionDelay = 1 * time.Second
+
+	// serviceRestartDelay is the delay between stop and start during restart
+	serviceRestartDelay = 2 * time.Second
+)
+
+type Tray struct {
+	// Menu items
+	menuOpenUI         *systray.MenuItem
+	menuStatus         *systray.MenuItem
+	menuIPFSStatus     *systray.MenuItem
+	menuPinShareStatus *systray.MenuItem
+	menuPeersStatus    *systray.MenuItem
+	menuSeparator1     *systray.MenuItem
+	menuStart          *systray.MenuItem
+	menuStop           *systray.MenuItem
+	menuRestart        *systray.MenuItem
+	menuSeparator2     *systray.MenuItem
+	menuSettings       *systray.MenuItem
+	menuLogs           *systray.MenuItem
+	menuAbout          *systray.MenuItem
+	menuSeparator3     *systray.MenuItem
+	menuExit           *systray.MenuItem
+
+	// State
+	serviceRunning bool
+	lastError      error
+}
+
+func NewTray() *Tray {
+	return &Tray{}
+}
+
+// BuildMenu creates the tray menu
+func (t *Tray) BuildMenu() {
+	// TODO: Re-enable when UI is ready
+	// // Open UI
+	// t.menuOpenUI = systray.AddMenuItem("Open PinShare UI", "Open the PinShare web interface")
+	//
+	// systray.AddSeparator()
+
+	// Status
+	t.menuStatus = systray.AddMenuItem("Status: Checking...", "Service status")
+	t.menuStatus.Disable()
+
+	t.menuIPFSStatus = systray.AddMenuItem("  IPFS: Unknown", "IPFS daemon status")
+	t.menuIPFSStatus.Disable()
+
+	t.menuPinShareStatus = systray.AddMenuItem("  PinShare: Unknown", "PinShare backend status")
+	t.menuPinShareStatus.Disable()
+
+	t.menuPeersStatus = systray.AddMenuItem("  Peers: Unknown", "Connected peers")
+	t.menuPeersStatus.Disable()
+
+	systray.AddSeparator()
+
+	// Service control
+	t.menuStart = systray.AddMenuItem("Start Service", "Start the PinShare service")
+	t.menuStop = systray.AddMenuItem("Stop Service", "Stop the PinShare service")
+	t.menuRestart = systray.AddMenuItem("Restart Service", "Restart the PinShare service")
+
+	systray.AddSeparator()
+
+	// Settings and logs
+	t.menuSettings = systray.AddMenuItem("Settings...", "Open settings")
+	t.menuLogs = systray.AddMenuItem("View Logs...", "Open log directory")
+
+	systray.AddSeparator()
+
+	// About
+	t.menuAbout = systray.AddMenuItem("About PinShare", "About this application")
+
+	systray.AddSeparator()
+
+	// Exit
+	t.menuExit = systray.AddMenuItem("Exit", "Exit the PinShare tray application")
+
+	// Handle menu clicks
+	go t.handleMenuClicks()
+
+	// Initial status check
+	t.updateStatus()
+}
+
+// handleMenuClicks handles menu item clicks
+func (t *Tray) handleMenuClicks() {
+	for {
+		select {
```

Comment:
I feel like this function should receive a ctx and the select should include a case for ctx.Done() in order to make sure the go routine exits gracefully..